### ISSUE TITLE
[ARM/Linux] SIGSEGV in ProfileEnter/ProfileLeave

### ISSUE THREAD

**Issue by viewizard** [dotnet/coreclr#26749]
Hello,
I faced with SIGSEGV in ProfileEnter/ProfileLeave if callbacks setted up by SetEnterLeaveFunctionHooks3WithInfo().
Event mask:
```
COR_PRF_MONITOR_ENTERLEAVE | COR_PRF_ENABLE_FUNCTION_ARGS |
COR_PRF_ENABLE_FUNCTION_RETVAL | COR_PRF_ENABLE_FRAME_INFO |
COR_PRF_ENABLE_STACK_SNAPSHOT | COR_PRF_MONITOR_CLASS_LOADS |
COR_PRF_ENABLE_OBJECT_ALLOCATED | COR_PRF_MONITOR_OBJECT_ALLOCATED | COR_PRF_MONITOR_GC
```
Profiling works fine, but on program close, I see SIGSEGVs in ProfileEnter/ProfileLeave like this one:
```
Thread 14 received signal SIGSEGV, Segmentation fault.
[Switching to LWP 1977]
ProfileLeave (clientData=<error reading variable: Cannot access memory at address 0xac7e456c>, platformSpecificHandle=0xac7e4640)
    at /usr/src/debug/coreclr-3.0.0/src/vm/proftoeeinterfaceimpl.cpp:10449
10456	/usr/src/debug/coreclr-3.0.0/src/vm/proftoeeinterfaceimpl.cpp: No such file or directory.
(gdb) bt
#0  ProfileLeave (clientData=<error reading variable: Cannot access memory at address 0xac7e456c>, platformSpecificHandle=0xac7e4640)
    at /usr/src/debug/coreclr-3.0.0/src/vm/proftoeeinterfaceimpl.cpp:10449
#1  0xb6fe29e8 in update_get_addr (ti=0x0) at dl-tls.c:802
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
```
I did try check ```pProfInterface``` status before real ```pProfInterface->GetEnter3WithInfoHook()``` call in ProfileEnter/ProfileLeave by
```
    if (g_profControlBlock.pProfInterface.Load() == NULL)
    {
         goto LExit;
    }
```
but rarely I see another SIGSEGV:
```
Thread 19 received signal SIGSEGV, Segmentation fault.
[Switching to LWP 2283]
ProfileEnter (clientData=<error reading variable: Cannot access memory at address 0xa6e8356c>, 
    platformSpecificHandle=<error reading variable: Cannot access memory at address 0xa6e83568>) at /usr/src/debug/coreclr-3.0.0/src/vm/proftoeeinterfaceimpl.cpp:10283
10283	/usr/src/debug/coreclr-3.0.0/src/vm/proftoeeinterfaceimpl.cpp: No such file or directory.
(gdb) bt
#0  ProfileEnter (clientData=<error reading variable: Cannot access memory at address 0xa6e8356c>, 
    platformSpecificHandle=<error reading variable: Cannot access memory at address 0xa6e83568>) at /usr/src/debug/coreclr-3.0.0/src/vm/proftoeeinterfaceimpl.cpp:10283
#1  0xb32860ec in GetThread () at /usr/src/debug/coreclr-3.0.0/src/vm/threads.inl:34
#2  SetCallbackStateFlagsHolder::SetCallbackStateFlagsHolder (dwFlags=3, this=<optimized out>) at /usr/src/debug/coreclr-3.0.0/src/vm/profilinghelper.inl:23
#3  ProfileEnter (
inline-frame.c:167: internal-error: void inline_frame_this_id(frame_info*, void**, frame_id*): Assertion `frame_id_p (*this_id)' failed.
```
looks like ```pProfInterface``` object was removed in another thread after my check.
Could you please help?

**Comment by alpencolt** [comment]
@jkotas PTAL

**Comment by jkotas** [comment]
Same underlying problem as https://github.com/dotnet/coreclr/issues/26687

cc @davmason @dotnet/dotnet-diag

**Comment by davmason** [comment]
Thanks for reporting this @viewizard. I will investigate and get back to you

**Comment by davmason** [comment]
I have a PR with a fix out in #26762, I am closing this issue as duplicate to #26687.
